src/imitation/algorithms/preference_comparisons.py

```diff
@@ -154,16 +166,26 @@ def __init__(
         # changes the reward function), so that we return the original environment
         # rewards.
         self.buffering_wrapper = wrappers.BufferingWrapper(venv)
-        self.venv = reward_wrapper.RewardVecEnvWrapper(
+        self.venv_train = reward_wrapper.RewardVecEnvWrapper(
             self.buffering_wrapper,
-            self.reward_fn,
+            reward_fn=self.reward_fn,
         )
-        self.log_callback = self.venv.make_log_callback()
+        self.venv_eval = reward_wrapper.RewardVecEnvWrapper(
```

Comment:
Looking at this more closely, do we actually need `venv_eval`? The only time we interact with the environment is https://github.com/HumanCompatibleAI/imitation/blob/6e3919b259ab75fe8aecc7906f9b238a5341d546/src/imitation/algorithms/preference_comparisons.py#L259 and https://github.com/HumanCompatibleAI/imitation/blob/6e3919b259ab75fe8aecc7906f9b238a5341d546/src/imitation/algorithms/preference_comparisons.py#L281 Critically, in both cases we don't make use of the reward in any way. So I think we could get away with just using the underlying `self.buffering_wrapper`. If so this would technically eliminate the need for even having `update_norm_stats` flag, although we might want to retain that feature anyway. What do you think?

Reply:
Good point! I wasn't even aware of this. The `TrajectoryGenerator` directly get trajectories from `self.buffering_wrapper`, there is no need to wrap it with a `RewardVecEnvWrapper` at all!

I think generally it's useful to keep that feature. By default people can still do `rollout.generate_trajectories(policy, rew_wrapped_env)` without noticing the effects on the norm stats! I can also imagine people want to use `rollout.generate_trajectories(policy, rew_wrapped_env)` to generate trajectories with learned rewards, for analysis purpose.

I think it's fine to just use `buffering_wrapper` to better signal the sampled trajectories contain the original rewards. I also wish to refactor `self.buffering_wrapper` as `self.buffer_wrapped_venv` or `self.buffering_venv`.